CMake/RealmCore.cmake

```diff
@@ -342,10 +378,15 @@ endfunction()
 
 function(clone_and_build_realm_sync branch)
     set(cmake_files ${CMAKE_CURRENT_BINARY_DIR}${CMAKE_FILES_DIRECTORY})
+    if(REALM_PLATFORM STREQUAL "Android")
+        set(config_cmd test -f src/config.mk || REALM_CORE_PREFIX=${cmake_files}/realm-core/src/realm-core REALM_FORCE_OPENSSL=YES REALM_ENABLE_ENCRYPTION=YES REALM_ENABLE_ASSERTIONS= sh build.sh config && echo "ENABLE_ENCRYPTION    = yes" >> src/config.mk)
```

Comment:
If `ENABLE_ENCRYPTION` isn't set, OpenSSL will _not_ be downloaded, built and included. `REALM_ENABLE_ENCRYPTION=YES` is not required (it is ignored by realm-sync's `build.sh`).